specs/_features/maxeb_increase/capella.py

```diff
@@ -1548,9 +1558,17 @@ def process_registry_updates(state: BeaconState) -> None:
         # Order by the sequence of activation_eligibility_epoch setting and then index
     ], key=lambda index: (state.validators[index].activation_eligibility_epoch, index))
     # Dequeued validators for activation up to churn limit
-    for index in activation_queue[:get_validator_churn_limit(state)]:
+    activation_balance_to_consume = get_validator_churn_limit(state)
+    for index in activation_queue:
         validator = state.validators[index]
-        validator.activation_epoch = compute_activation_exit_epoch(get_current_epoch(state))
+        # Validator can now be activated
+        if state.activation_validator_balance + activation_balance_to_consume >= validator.effective_balance:
+            activation_balance_to_consume -= (validator.effective_balance - state.activation_validator_balance)
+            state.activation_validator_balance = Gwei(0)
+            validator.activation_epoch = compute_activation_exit_epoch(get_current_epoch(state))
+        else:  
+            state.activation_validator_balance += activation_balance_to_consume
+            break
```

Comment:
Can this be simplified to the following?

```python
    state.activation_queue_budget += get_validator_churn_limit(state)
    for index in activation_queue:
        validator = state.validators[index]
        if validator.effective_balance <= state.activation_queue_budget:
            state.activation_queue_budget -= validator.effective_balance
            validator.activation_epoch = compute_activation_exit_epoch(get_current_epoch(state))
```

Reply:
There’s a problem with this. Suppose that each epoch has a single 32 ETH validator in the activation queue. Then, state.activation_churn_credit is increased by (churn_limit - 32) per epoch. Later on, state.activation_churn_credit can be so large that a significant fraction of the validator set can exit at once.

Reply:
@michaelneuder Francesco is right, the logic is wrong 🙈

Reply:
aha, good point! rolled it back to my previous version :-)